tests/unit/s2n_prf_key_material_test.c

```diff
@@ -0,0 +1,235 @@
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include "s2n_test.h"
+#include "testlib/s2n_testlib.h"
+#include "tls/s2n_prf.h"
+#include "utils/s2n_random.h"
+
+static S2N_RESULT s2n_test_validate_key_material(struct s2n_key_material *key_material,
+        struct s2n_blob *test_data_blob, uint8_t mac_size, uint8_t key_size, uint8_t iv_size)
+{
+    /* confirm that the data is copied to key_material */
+    RESULT_ENSURE_EQ(test_data_blob->size, sizeof(key_material->key_block));
+    RESULT_ENSURE_EQ(memcmp(test_data_blob->data, key_material->key_block, test_data_blob->size), 0);
+
+    struct s2n_stuffer test_data_stuffer = { 0 };
+    RESULT_GUARD_POSIX(s2n_stuffer_init_written(&test_data_stuffer, test_data_blob));
+
+    /* test that its possible to access data from s2n_key_material */
+    /* client MAC */
+    uint8_t *test_ptr = s2n_stuffer_raw_read(&test_data_stuffer, mac_size);
+    RESULT_ENSURE_REF(test_ptr);
+    RESULT_ENSURE_EQ(memcmp(test_ptr, key_material->client_mac.data, mac_size), 0);
+    /* server MAC */
+    test_ptr = s2n_stuffer_raw_read(&test_data_stuffer, mac_size);
+    RESULT_ENSURE_REF(test_ptr);
+    RESULT_ENSURE_EQ(memcmp(test_ptr, key_material->server_mac.data, mac_size), 0);
+
+    /* client KEY */
+    test_ptr = s2n_stuffer_raw_read(&test_data_stuffer, key_size);
+    RESULT_ENSURE_REF(test_ptr);
+    RESULT_ENSURE_EQ(memcmp(test_ptr, key_material->client_key.data, key_size), 0);
+    /* server KEY */
+    test_ptr = s2n_stuffer_raw_read(&test_data_stuffer, key_size);
+    RESULT_ENSURE_REF(test_ptr);
+    RESULT_ENSURE_EQ(memcmp(test_ptr, key_material->server_key.data, key_size), 0);
+
+    /* client IV */
+    test_ptr = s2n_stuffer_raw_read(&test_data_stuffer, iv_size);
+    RESULT_ENSURE_REF(test_ptr);
+    RESULT_ENSURE_EQ(memcmp(test_ptr, key_material->client_iv.data, iv_size), 0);
+    /* server IV */
+    test_ptr = s2n_stuffer_raw_read(&test_data_stuffer, iv_size);
+    RESULT_ENSURE_REF(test_ptr);
+    RESULT_ENSURE_EQ(memcmp(test_ptr, key_material->server_iv.data, iv_size), 0);
+
+    return S2N_RESULT_OK;
+}
+
+int main(int argc, char **argv)
+{
+    BEGIN_TEST();
+
+    /* prepare test data */
+    uint8_t test_data[S2N_MAX_KEY_BLOCK_LEN] = { 0 };
+    struct s2n_blob test_data_blob = { 0 };
+    EXPECT_SUCCESS(s2n_blob_init(&test_data_blob, test_data, sizeof(test_data)));
+    EXPECT_OK(s2n_get_public_random_data(&test_data_blob));
+
+    /* fuzz s2n_key_material_init with different mac, key, iv sizes */
+    {
+        for (uint8_t mac_size = 0; mac_size < SHA512_DIGEST_LENGTH; mac_size++) {
```

Comment:
Okay, just style nits now :)

I generally don't bother with the enclosing { } if the test starts with a loop, but that's not a hard rule.
We also generally put a `;` at the end of the test { }, because a quirk of clang-format requires it in some cases.